shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -341,23 +341,38 @@ - (void)updateViewportMetrics {
   });
 }
 
+- (bool)isWindowFullscreen {
+  UIWindow *window = self.view.window;
+  return CGRectEqualToRect(window.frame, window.screen.bounds);
+}
+
+- (CGFloat)statusBarPadding {
+  // If we're a child of a containing view, let the container apply padding.
+  if (self.parentViewController != nil) {
+    return 0.0;
+  }
```

Comment:
> Any case where this isn't a legitimate assumption?

I am uncertain. I'd keep this check in place. In the future, if embedders want to have more control over this behavior, we can add a `FlutterViewControllerDelegate` and ask it the embedder is handling status bar behavior or us. I feel like we have already covered all our bases with this patch.